### ISSUE TITLE
Athena error

### ISSUE THREAD

**Issue by kleijnweb** [johanneswuerbach/jaeger-s3#350]
Hi Johannes, any idea how to debug this?

2023-08-02T08:45:20.074Z [ERROR] jaeger-s3: failed to get dependencies: @module=jaeger-s3 EXTRA_VALUE_AT_END="failed to query athena: failed to get athena query result: operation error Athena: GetQueryResults, https response error StatusCode: 400, RequestID: 0fe475ae-04a0-4641-9607-e004d5aa3e6a, InvalidRequestException: Query did not finish successfully. Final query state: FAILED" timestamp=2023-08-02T08:45:20.073Z

**Comment by kleijnweb** [comment]
For context, everything else seems to be working fine. Operator is deployed correctly, sidecars injected. Both the main and results buckets are being populated with expected data. It's just that I can't query the results using the UI.

**Comment by kleijnweb** [comment]
If I execute this query in the AWS console, it just works, so it doesn't seem like it's an issue with the query used by GetDependencies():

```
WITH spans_with_references AS (
			SELECT
				base.service_name,
				base.trace_id,
				base.span_id,
				unnested_references.reference.trace_id as ref_trace_id,
				unnested_references.reference.span_id as ref_span_id
			FROM "jaeger_spans" as base
			CROSS JOIN UNNEST(base.references) AS unnested_references (reference)
		)

		SELECT jaeger.service_name as parent, spans_with_references.service_name as child, COUNT(*) as callcount
			FROM spans_with_references
			JOIN "jaeger_spans" as jaeger ON spans_with_references.ref_trace_id = jaeger.trace_id AND spans_with_references.ref_span_id = jaeger.span_id
			WHERE datehour BETWEEN '2023/08/01/00' and '2023/08/02/14'
			GROUP BY 1, 2
```

**Comment by kleijnweb** [comment]
This is my first time doing anything with Athena and turns out I can just lookup the failed queries. Apparently the error is:

```
HIVE_CURSOR_ERROR: Failed to read Parquet file: s3://test-jaeger/spans/2023/08/02/13/oHqdEIWOrrnrQaymuxIGyruJzdxWaKlc.parquet
```

**Comment by kleijnweb** [comment]
After destroying and recreating all resources, everything seems to work as expected. My working theory is that because the permissions were only partially correct initially on Wednesday morning, there was some data integrity issue between the different data sources (s3, glue, Athena).
Hopefully the issue will not recur, but for now it seems solved.

**Comment by kleijnweb** [comment]
Alright the issue recurred. [This link](https://repost.aws/questions/QU5IM2o3sSSGSH-Yw8fiXKsw/hive-cursor-error-failed-to-read-parquet-file) suggests that it may be caused by a DDL mismatch, but the weird thing is it was working this morning. Not to mention those integration tests passed, and the failed query does not seem related to GetDependencies().

**Comment by kleijnweb** [comment]
Turns out the issue was with the version of Athena, TF creates a v2 by default but auto upgrading caused it to be running v2 a little later, which doesn't work.

Solved it (sort of, would be better if it worked with v3) by adding this to the TF:

```
  configuration {
    engine_version {
     selected_engine_version = "Athena engine version 2"
    }
 }
```

Perhaps worth adding to the setup guide.

**Comment by johanneswuerbach** [comment]
Thank you @kleijnweb, happy to review a PR with those changes.

> would be better if it worked with v3

Agreed, but I haven't had time yet to look into what changes would be required.

**Comment by kleijnweb** [comment]
I'll create a PR. Question: is there any reason bucket versioning is explicitly enabled in the terraform?

**Comment by johanneswuerbach** [comment]
I don't remember, but I don't think it is required.

**Comment by alexf101** [comment]
Hey, we had the same problem after upgrading to v3, and were lucky enough to get a helpful suggestion from our account manager that apparently the problem has something to do with a feature in v3 that tries to reduce the number of files scanned by reading some metadata. She suggested turning it off by adding the following to the `create table`:

`WITH SERDEPROPERTIES ('parquet.ignore.statistics'='true') `

This worked for us, though I had to recreate the table - couldn't figure out a syntax to do this with an `alter`, maybe I missed something obvious though.

Hope this helps, but of course it begs the question of how to take advantage of the new metadata thing; I think it's this? https://prestodb.io/blog/2022/05/10/faster-presto-queries-with-parquet-page-index#parquet-statistics

I'm way out of my depth now though. It would be really cool though if we could accelerate finding queries by `traceId` - AFAICT, if we used a timestamp at the front of a traceId (like a Mongo [ObjectIds](https://www.mongodb.com/docs/manual/reference/method/ObjectId/)), it should be possible to show Athena which range of traceIds are in each file? Is that... at all how this works?
